File: CS/Sample/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using Sample.Models;

namespace Sample.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();

        public ActionResult Index() {
            return View(list.GetPersons());
        }

        public ActionResult GridViewEditingPartial() {
            return PartialView(list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingAddNew([ModelBinder(typeof(DevExpressEditorsBinder))] Person person, bool keepOpened) {
            if(ModelState.IsValid)
                list.AddPerson(person);

            ViewData["KeepOpened"] = keepOpened;
            ViewData["EditingRowKey"] = person.PersonID;
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingUpdate([ModelBinder(typeof(DevExpressEditorsBinder))] Person personInfo, bool keepOpened) {
            if(ModelState.IsValid)
                list.UpdatePerson(personInfo);

            ViewData["KeepOpened"] = keepOpened;
            ViewData["EditingRowKey"] = personInfo.PersonID;
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }

        public ActionResult EditingDelete(int personId) {
            list.DeletePerson(personId);
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }
    }
}

File: VB/Sample/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports Sample.Models

Namespace Sample.Controllers
	Public Class HomeController
		Inherits Controller
		Private list As New PersonsList()

		Public Function Index() As ActionResult
			Return View(list.GetPersons())
		End Function

		Public Function GridViewEditingPartial() As ActionResult
			Return PartialView(list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)> _
		Public Function EditingAddNew(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal person As Person, ByVal keepOpened As Boolean) As ActionResult
			If ModelState.IsValid Then
				list.AddPerson(person)
			End If

			ViewData("KeepOpened") = keepOpened
			ViewData("EditingRowKey") = person.PersonID
			Return PartialView("GridViewEditingPartial", list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)> _
		Public Function EditingUpdate(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal personInfo As Person, ByVal keepOpened As Boolean) As ActionResult
			If ModelState.IsValid Then
				list.UpdatePerson(personInfo)
			End If

			ViewData("KeepOpened") = keepOpened
			ViewData("EditingRowKey") = personInfo.PersonID
			Return PartialView("GridViewEditingPartial", list.GetPersons())
		End Function

		Public Function EditingDelete(ByVal personId As Integer) As ActionResult
			list.DeletePerson(personId)
			Return PartialView("GridViewEditingPartial", list.GetPersons())
		End Function
	End Class
End Namespace